src/main/java/dev/meyi/bn/commands/BazaarNotifierCommand.java

```diff
@@ -1,178 +1,275 @@
+                  : EnumChatFormatting.RED) + "The mod has been toggled " + (
+                  BazaarNotifier.activeBazaar ? EnumChatFormatting.DARK_GREEN + "on"
+                      : EnumChatFormatting.DARK_RED + "off")));
+        } else {
+          player.addChatMessage(new ChatComponentText(
+              BazaarNotifier.prefix + EnumChatFormatting.RED
+                  + "Run /bn api (key) to set your api key. Do /api if you need to get your api key"));
+        }
+      } else if (args.length >= 1 && args[0].equalsIgnoreCase("api")) {
+        if (args.length == 2) {
+          BazaarNotifier.apiKey = args[1];
+          try {
+            if (Utils.validateApiKey()) {
+              player.addChatMessage(new ChatComponentText(
+                  BazaarNotifier.prefix + EnumChatFormatting.RED
+                      + "Your api key has been set."));
+              BazaarNotifier.apiKey = "";
+              BazaarNotifier.validApiKey = true;
+              BazaarNotifier.activeBazaar = true;
+            } else {
+              player.addChatMessage(new ChatComponentText(
+                  BazaarNotifier.prefix + EnumChatFormatting.RED
+                      + "Your api key is invalid. Please run /api new to get a fresh api key & use that in /bn api (key)"));
+              BazaarNotifier.validApiKey = false;
+            }
+          } catch (IOException e) {
+            player.addChatMessage(new ChatComponentText(
+                BazaarNotifier.prefix + EnumChatFormatting.RED
+                    + "An error occurred when trying to set your api key. Please re-run the command to try again."));
+            BazaarNotifier.validApiKey = false;
+            e.printStackTrace();
+          }
+        } else {
+          player.addChatMessage(new ChatComponentText(
+              BazaarNotifier.prefix + EnumChatFormatting.RED
+                  + "Run /bn api (key) to set your api key. Do /api if you need to get your api key."));
+          BazaarNotifier.validApiKey = false;
+        }
+      } else if (args.length == 1 && args[0].equalsIgnoreCase("dump")) {
+        System.out.println(BazaarNotifier.orders);
+        player.addChatMessage(new ChatComponentText(BazaarNotifier.prefix + EnumChatFormatting.RED
+            + "Orders dumped to the log file"));
+      } else if (args.length == 2 && args[0].equalsIgnoreCase("reset")) {
+        if (args[1].equalsIgnoreCase("all")) {
+          BazaarNotifier.resetMod();
+          player.addChatMessage(new ChatComponentText(BazaarNotifier.prefix + EnumChatFormatting.RED
+              + "All module locations have been reset and the order list has been emptied"));
+        } else if (args[1].equalsIgnoreCase("orders")) {
+          BazaarNotifier.orders = Defaults.DEFAULT_ORDERS_LAYOUT();
+          player.addChatMessage(new ChatComponentText(BazaarNotifier.prefix + EnumChatFormatting.RED
+              + "Your orders have been cleared"));
+        }

```

Comment:
Let's revert this change. Right now, this means `/bn reset` isn't a valid command but `/bn reset _____` is.

https://github.com/symt/BazaarNotifier/pull/49/files#diff-b2e22e17a0a577723425dcf654ec43f0059ca89f2fef97da4f87a62e669cf13eL93-L102

Reply:
I thought its confusing to have a 3rd optional argument that gets recommended.